# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1475782866 14400
#      Thu Oct 06 15:41:06 2016 -0400
# Branch stable
# Node ID 610f88ed2b789f9f9bdb648a47460fa980a64dbb
# Parent  883b72cdf0b2563cae59c973e46d729df1e69b8f
avoid gnulib waitpid replacement for Windows systems (bug #49211)

* wait-wrappers.c (octave_waitpid_wrapper): Always set status to 0 and
return -1 on Windows systems.

diff --git a/liboctave/wrappers/wait-wrappers.c b/liboctave/wrappers/wait-wrappers.c
--- a/liboctave/wrappers/wait-wrappers.c
+++ b/liboctave/wrappers/wait-wrappers.c
@@ -44,17 +44,24 @@ along with Octave; see the file COPYING.
 
 #if ! defined (WUNTRACED)
 #  define WUNTRACED 0
 #endif
 
 pid_t
 octave_waitpid_wrapper (pid_t pid, int *statusp, int options)
 {
+#if defined (__WIN32__) && ! defined (__CYGWIN__)
+  // gnulib's waitpid replacement currently uses _cwait, which
+  // apparently only works with console applications.
+  *statusp = 0;
+  return -1;
+#else
   return waitpid (pid, statusp, options);
+#endif
 }
 
 int
 octave_wcontinue_wrapper (void)
 {
   return WCONTINUE;
 }
 
